ociclient/ociclient.go

```diff
@@ -0,0 +1,152 @@
+	oc.vc = &vaultsClient
+
+	log.Info("ociclient/ociclient:InitializeClient() OCI client initialized")
+
+	return nil
+}
+
+func (oc *ociClient) CreateKey(OciCompartmentId, OciKeyId, OciSecretName, OciVaultId string, OciSecretId *string) error {
+	// Create request and dependent object(s).
+	req := vault.CreateSecretRequest{
+		CreateSecretDetails: vault.CreateSecretDetails{
+			// Required
+			CompartmentId: &OciCompartmentId,
+			KeyId:         &OciKeyId,
+			SecretName:    &OciSecretName,
+			VaultId:       &OciVaultId,
+
+			// TODO: For now just use random bytes, but it should be possible
+			//       to select the key type.
+			SecretGenerationContext: vault.BytesGenerationContext{
+				GenerationTemplate: vault.BytesGenerationContextGenerationTemplate512,
+			},
+
+			// Optional
+			EnableAutoGeneration: common.Bool(true),
+		},
+	}
+
+	// Send the request using the vault client.
+	resp, err := oc.vc.CreateSecret(context.Background(), req)
+	if err != nil {
+		return errors.Wrap(err, "ociclient/ociclient:CreateKey() Failed to create OCI secret")
+	}
+
+	// Retrieve value from the response.
+	*OciSecretId = *resp.Secret.Id
+
+	log.Infof("ociclient/ociclient:CreateKey() Created key '%s' (%s) on oci server", *OciSecretId, OciSecretName)
+
+	return nil
+}
+
+func (oc *ociClient) DeleteKey(secretId string) error {
+	// Create a request and dependent object(s).
+	req := vault.ScheduleSecretDeletionRequest{
+		SecretId: common.String(secretId),
+
+		ScheduleSecretDeletionDetails: vault.ScheduleSecretDeletionDetails{
+			TimeOfDeletion: &common.SDKTime{
+				Time: time.Now().Add(time.Hour * 48),
+			},
+		},
+	}
+
+	// Send the request using the vault client.
+	resp, err := oc.vc.ScheduleSecretDeletion(context.Background(), req)
+	if err != nil {
+		return errors.Wrapf(err, "Failed to delete key '%s' from oci server", secretId)
+	}
+
+	// TODO: Check response for error.
+	fmt.Printf("ociclient/ociclient:DeleteKey(): response:\n%s\n\n", resp)
+
+	log.Infof("ociclient/ociclient:DeleteKey() Deleted key '%s' from oci server", secretId)
+
+	return nil
+}
+
+func (oc *ociClient) GetKey(secretId string, secretVersionNumber int64) ([]byte, error) {
+	// Create request and dependent object(s).
+	req := secrets.GetSecretBundleRequest{
+		SecretId: common.String(secretId),
+	}
+
+	// If the secret version was provided, include it. Otherwise it'll just
+	// fallback to the current version.
+	if secretVersionNumber > 0 {
+		req.VersionNumber = common.Int64(secretVersionNumber)
+	}
+
+	// Send the request using the secrets client.
+	resp, err := oc.sc.GetSecretBundle(context.Background(), req)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get secret bundle")
+	}
+
+	// Parse bundle contents.
+	re := regexp.MustCompile(`\{ Content=([-A-Za-z0-9+/]*) \}`)
+	strs := re.FindStringSubmatch(fmt.Sprintf("%s", resp.SecretBundle.SecretBundleContent))
+	if strs == nil || len(strs) != 2 {
+		return nil, errors.New("could not match regular expression")
```

Comment:
can we improve error message? `could not parse secret bundle` or `could not extract secret`?

Reply:
Will do.